src/data/polynomial/degree/reverse_mul.lean

```diff
@@ -0,0 +1,605 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+
+import tactic
+import data.polynomial.degree.basic
+import data.polynomial.degree.trailing_degree
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+lemma nonempty_of_card_nonzero {α : Type*} {s : finset α} : s.card ≠ 0 → s.nonempty :=
+begin
+  contrapose!,
+  rw [nonempty_iff_ne_empty, not_not, card_eq_zero, imp_self],
+  trivial,
+end
+
+lemma nonempty_of_card_le_two {α : Type*} {s : finset α} (H : 2 ≤ s.card) : s.nonempty :=
+begin
+  apply nonempty_of_card_nonzero,
+  intro,
+  rw [a, nat.le_zero_iff] at H,
+  revert H,
+  exact two_ne_zero,
+end
+
+lemma card_two {sup : finset ℕ} (H : 2 ≤ sup.card) : finset.min' sup ≠ finset.max' sup :=
```

Comment:
This is mostly `finset.min'_lt_max'_of_card`, so if you really need the `ne` version of that as a separate lemma, you should still make it as close to `finset.min'_lt_max'_of_card` as possible. In particular,

- Put it in the same place, so someone browsing the docs can see both
- Give them analogous names, like `finset.min'_ne_max'_of_card`
- Make them use the same variables